Src/Particle/AMReX_ParticleMesh.H

```diff
@@ -14,11 +14,10 @@ ParticleToMesh (PC const& pc, MF& mf, int lev, F&& f)
 {
     BL_PROFILE("amrex::ParticleToMesh");
 
-    MultiFab* mf_pointer = pc.OnSameGrids(lev, mf) ?
-        &mf : new MultiFab(pc.ParticleBoxArray(lev),
-                           pc.ParticleDistributionMap(lev),
-                           mf.nComp(), mf.nGrowVect());
-    mf_pointer->setVal(0.);
+    MultiFab mf_tmp(pc.ParticleBoxArray(lev),
+                    pc.ParticleDistributionMap(lev),
+                    mf.nComp(), mf.nGrowVect());
+    mf_tmp.setVal(0.0);
```

Comment:
Offline discussed: I was wondering if an additional flag could save the temporary multifab if we pass multiple `pc`s

Reply:
Not worried about temporary multifabs.  But it would certainly save communications.

Reply:
It seems that if 1) the input multifab has zero ghost cells (it doesn't need any for NGP deposition) and 2) dual grid is off, we could skip the temporary altogether. That would cover the "particle_count" case anyway.